pkg/core/resources/manager/manager.go

```diff
@@ -130,7 +130,7 @@ var ErrSkipUpsert = errors.New("don't do upsert")
 func Upsert(ctx context.Context, manager ResourceManager, key model.ResourceKey, resource model.Resource, fn func(resource model.Resource) error, fs ...UpsertFunc) error {
 	upsert := func(ctx context.Context) error {
 		create := false
-		err := manager.Get(ctx, resource, store.GetBy(key))
+		err := manager.Get(ctx, resource, store.GetBy(key), store.GetConsistent())
```

Comment:
Wouldn't it be better to utilize postgres `insert ... on conflict` clause instead of reimplementing it as `upsert` in our code? With `on conflict` we are always performing operations on master

Reply:
you can specify `conflict-target` so it does not only work with primary keys

Reply:
I am not sure, but probably conflict target needs to be unique, so in our case it won;t work